<launch>
    <!-- Task and logging arguments -->
    <arg name="task_type"/>
    <arg name="logging_enabled"                 default="true"/>
    <arg name="test_id"                         default=""/>
    <arg name="screenshots_enabled"             default="false"/>

    <!-- Simulator arguments -->
    <arg name="feedback_covariance"             default="0.0"/>
    <arg name="start_bullet_viewer"             default="true"/>

    <!-- Planner model arguments -->
    <arg name="bandit_algorithm"                default="KFRDB"/>
    <arg name="multi_model"                     default="false"/>
    <arg name="deformability_override"          default="false"/>
    <arg if="$(arg deformability_override)"     name="translational_deformability"/>
    <arg if="$(arg deformability_override)"     name="rotational_deformability"/>
    <arg name="use_adaptive_model"              default="false"/>
    <arg name="adaptive_model_learning_rate"    default="0.000001"/>
    <arg name="calculate_regret"                default="false"/>

    <!-- Planner internal arguments -->
    <arg name="planning_horizion"               default="1"/>
    <arg name="kalman_parameters_override"      default="false"/>

    <!-- Launch the simulator and setup parameters -->
    <include file="$(find smmap)/launch/generic_experiment_sim_only.launch">
        <!-- Task and logging arguments -->
        <arg name="task_type"                       value="$(arg task_type)"/>
        <arg name="logging_enabled"                 value="$(arg logging_enabled)"/>
        <arg name="test_id"                         value="$(arg test_id)"/>
        <arg name="screenshots_enabled"             value="$(arg screenshots_enabled)"/>

        <!-- Simulator arguments -->
        <arg name="feedback_covariance"             value="$(arg feedback_covariance)"/>
        <arg name="start_bullet_viewer"             value="$(arg start_bullet_viewer)"/>

        <!-- Planner model arguments -->
        <arg name="deformability_override"          value="$(arg deformability_override)"/>
        <arg name="translational_deformability"     value="$(arg translational_deformability)"  if="$(arg deformability_override)"/>
        <arg name="rotational_deformability"        value="$(arg rotational_deformability)"     if="$(arg deformability_override)"/>
        <arg name="multi_model"                     value="$(arg multi_model)"/>
        <arg name="use_adaptive_model"              value="$(arg use_adaptive_model)"/>
        <arg name="adaptive_model_learning_rate"    value="$(arg adaptive_model_learning_rate)"/>
        <arg name="calculate_regret"                value="$(arg calculate_regret)"/>

        <!-- Planner internal arguments -->
        <arg name="planning_horizion"               value="$(arg planning_horizion)"/>
        <arg name="kalman_parameters_override"      value="$(arg kalman_parameters_override)"/>
        <arg name="process_noise_factor"            value="$(arg process_noise_factor)"         if="$(arg kalman_parameters_override)"/>
        <arg name="observation_noise_factor"        value="$(arg observation_noise_factor)"     if="$(arg kalman_parameters_override)"/>
    </include>

    <!-- Start the planner -->
    <node name="smmap_planner_node" pkg="smmap" type="smmap_test_node_$(arg bandit_algorithm)" required="true"/>
</launch>
